configurer/linux.go

```diff
@@ -87,6 +88,25 @@ func (l Linux) K0sJoinTokenPath() string {
 	return "/etc/k0s/k0stoken"
 }
 
+// TryLock tries to obtain an exclusive lock on the host to avoid multiple instances accessing it at once
+func (l Linux) TryLock(h os.Host) error {
+	if err := h.Exec("command -v flock"); err != nil {
+		log.Warnf("%s: host does not have the 'flock' command which is used to ensure only one instance of k0sctl operates on it at a time", h)
+		return nil
+	}
+	sshpid, err := h.ExecOutput("ps --no-headers -eo ppid -fp $$")
+	if err != nil {
+		return err
+	}
+	if err := h.Execf(`timeout 10 flock /tmp/k0sctl.lock -c true`, exec.Sudo(h)); err != nil {
+		return fmt.Errorf("failed to obtain a lock for host")
+	}
+	go func() {
+		_ = h.Execf(`flock -n /tmp/k0sctl.lock -c "while ps -p %s > /dev/null; do sleep 1; done;"`, sshpid, exec.Sudo(h))
+	}()
+	return nil
```

Comment:
not sure I read this right:
1. we try to acquire the lock and fail if we do not get it in 10secs
2. we start go routine to keep the lock untill the parent ssh pid is gone

Isn't there slight window between 1 and 2 where someone else could possibly get the lock?

I'm not foo familiar with `flock`, can't it be used like:
1. get the lock
2. do stuff
3. release the lock

essentially step 3 would be the "cleanup" at disconnecting from the node

Reply:
Yeah there's a window. I guess the window could be sealed by creating something that communicates the exec result from inside the goroutine to main if it exits early.

No need for cleanup if flock keeps the lock and releases it when the connection dies, it will cause much more headaches if killing k0sctl leaves behind open locks when cleanup doesn't happen.

Reply:
Window sealed, but it brings a mandatory delay of a couple of seconds on success.